### PR TITLE
Feat: delete tricks

### PR DESCRIPTION
**Pull request by niccoloMG**
## Description

Add a simple button on trick cards to trigger delete. Delete method automatically renders loading screen and fetches the updated trick list on success.

## Task

[Notion Task Link](https://www.notion.so/d077118807034bbfb517f36c20d2a304?v=345833d9689145a2b5a197646677c93e&p=00d56dc1bb1b4691b5ab220e5fcf0949&pm=s)

## Screenshots (optional)

image

### CODE REVIEW

File: src/components/Card.svelte [hunk 1]
```diff
@@ -16,6 +17,10 @@
 			text: trick.mix
 		}
 	];
+
+	const handleDelete = async() => {
+		await deleteTrick(trick.id)
```
Comment (tonyjojo) [inline src/components/Card.svelte:22]: Do we need the `await` here?

**Review by tonyjojo — APPROVED** [review]
Good stuff

File: src/components/Card.svelte (hunk 1 above)
Comment (berk245) [reply src/components/Card.svelte:22]: I think adding a confirm here would be nice to prevent accidental deletions

File: src/lib/handlers/db.js [hunk 2]
```diff
@@ -1,5 +1,11 @@
 import { db } from '$lib/firebase/firebase';
-import { collection, doc, getDoc, getDocs, setDoc } from 'firebase/firestore';
+import { collection, doc, getDoc, getDocs, setDoc, deleteDoc } from 'firebase/firestore';
+import { authStore, setLoading } from '../store/store';
+
+let user;
+authStore.subscribe((curr) => {
+	user = curr?.user;
+});
```
Comment (berk245) [inline src/lib/handlers/db.js:9]: This is fine for now, but we will probably need to think about abstracting getter methods from store

File: src/lib/handlers/db.js [hunk 3]
```diff
@@ -33,3 +39,26 @@ const createUserDocument = async (user) => {
 	console.log('User created');
 	return;
 };
+
+export const deleteTrick = async (trickId) => {
+	try {
+		setLoading(true)
+		const userRef = doc(db, 'users', user.uid);
+		const tricksCollection = collection(userRef, 'tricks');
```
Comment (berk245) [inline src/lib/handlers/db.js:49]: This is a repetition of the code in the main layout. We should create a function to get trick collection of a user

**Review by berk245 — CHANGES_REQUESTED** [review]
Let's add the confirm deletion before merging. The other points could be subject of a refactoring PR

File: src/components/Card.svelte (hunk 1 above)
Comment (niccoloMG) [reply src/components/Card.svelte:22]: Okay. I will add the confirm. I will also give feedback for loading and success/fail, so awaiting will be useful

File: src/lib/handlers/db.js (hunk 2 above)
Comment (niccoloMG) [reply src/lib/handlers/db.js:9]: Makes sense. Thanks for pointing it out.

File: src/components/Card.svelte (hunk 1 above)
Comment (niccoloMG) [reply src/components/Card.svelte:22]: Added confirm and alert after failed deletions

**Review by berk245 — APPROVED** [review]
Thanks for the changes

**Pull request merged**